src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -28,7 +28,7 @@ function WebGLShadowMap( _renderer, _lights, _objects, capabilities ) {
 	_maxShadowMapSize = new Vector2( capabilities.maxTextureSize, capabilities.maxTextureSize ),
 
 	_lookTarget = new Vector3(),
-	_lightPositionWorld = new Vector3(),
+	_lightPositionWorldAndScale = new Vector4(),
```

Comment:
Why are you packing it into a Vector4() here.  This can lead to confusing code and I am not sure why you are doing this here -- is the reason to save a uniform?  Are we at risk of running out of uniforms in this particular case?  I suspect not.